### ISSUE TITLE
SSL cert hot reloading

### ISSUE THREAD

**Issue by maxweisel** [verdaccio/verdaccio#3024]
**Is your feature request related to a problem?**
I use verdaccio with letsencrypt. I love using the pre-built docker image and I don't have to maintain a fork. One annoyance is that letsencrypt certificates expire every three months. At the moment I need to restart the container on a regular basis, but I'd rather have verdaccio regularly reload the SSL certificate without restarting the server.

**Describe the solution you'd like**
Node's https has a method called `setSecureContext` which you can use to update the cert/key on a running instance without restarting it. I'd love to either have a config option to let verdaccio replace this on a regular basis, or even better would be to watch the config file on disk and automatically reload whenever it changes.

**Describe alternatives you've considered**
The current alternative is to restart the server regularly which is a pain and results in downtime.

**Comment by juanpicado** [comment]
It has already been tried https://github.com/verdaccio/verdaccio/pull/1775

**Comment by maxweisel** [comment]
Dang, yeah very close, but certainly having to handle this for every request doesn't seem very efficient. If I get some time this week, I'll see if I can open a PR that implements it as described above. In theory this shouldn't introduce any overhead per-request and will only call setSecureContext when the certificate changes on disk.

**Comment by edclement** [comment]
Have you thought about putting an nginx or other reverse proxy in front of your Verdaccio instance? nginx supports hot reload of it's configuration without any interruption to service (including SSL certs). It might be better handled that way.

**Comment by maxweisel** [comment]
Yes, and our load balancers can do SSL termination too, but I don't want to do that here.
